fvm/storage_capacity_resolver.go

```diff
@@ -0,0 +1,132 @@
+package fvm
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type StorageCapacityResolver func(state.Ledger, flow.Address, Context) (uint64, error)
+
+func ResourceStorageCapacityResolver(l state.Ledger, address flow.Address, ctx Context) (uint64, error) {
+	storageReservation, err := getAccountsStorageReservation(l, address, ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	minimumStorageReservation, bytesPerFLOW, err := getStorageFeeParameters(l, ctx)
+	if err != nil {
+		return 0, err
+	}
+	if storageReservation < minimumStorageReservation {
+		return uint64(0), nil
+	}
+
+	storageCapacityValue := storageReservation.Mul(bytesPerFLOW).(interpreter.UFix64Value).ToInt()
+	return uint64(storageCapacityValue), nil
+}
+
+func getStorageFeeParameters(l state.Ledger, ctx Context) (interpreter.UFix64Value, interpreter.UFix64Value, error) {
+	flowStorageFeesId := flow.RegisterID{
+		Owner:      string(ctx.Chain.ServiceAddress().Bytes()),
+		Controller: "",
+		Key:        fmt.Sprintf("contract\x1F%s", "FlowStorageFees"),
```

Comment:
It would be good to use Cadence's function here instead of replicating its behaviour